### PR TITLE
O11Y-3138: Remove Deprecated SDK exports for SpanContext, TraceState, and W3cTraceContextPropagator

### PR DESCRIPTION
**Pull request by michaelyeager-wf**
## Which problem is this PR solving?

This PR removes deprecated exports for SpanContext, TraceState, and W3cTraceContextPropagator.  These classes were moved to API in https://github.com/Workiva/opentelemetry-dart/pull/109.

## How Has This Been Tested?

* Verified unit tests pass successfully following this change.

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 Recommendation: **:bangbang: Major version bump (fyi @Workiva/semver-audit-group )**


```diff
@@ opentelemetry/sdk.dart <-- package:opentelemetry/src/sdk/trace/trace_state.dart#L13 @@
-  class TraceState implements TraceState
// Removing a class is a major change.

-     TraceState TraceState.empty()
//    Removing a constructor is a major change.

-     int get size
//    Removing a field is a major change.

-     TraceState TraceState.fromString(String traceState)
//    Removing a constructor is a major change.

-     String toString()
//    Removing a method is a major change.

-     bool get isEmpty
//    Removing a field is a major change.

-     void put(String key, String value)
//    Removing a method is a major change.

-     void remove(String key)
//    Removing a method is a major change.

-     static RegExp get validKeyRegex
//    Removing a field is a major change.

-     String get(String key)
//    Removing a method is a major change.

-     static RegExp get validValueRegex
//    Removing a field is a major change.

-     static TraceState getDefault()
//    Removing a method is a major change.

      From dart:core/object.dart
-     dynamic noSuchMethod(Invocation invocation)
//    Removing a method is a major change.
-     int get hashCode
//    Removing a field is a major change.
-     bool ==(Object other)
//    Removing a method is a major change.
-     Type get runtimeType
//    Removing a field is a major change.

```

```diff
@@ opentelemetry/sdk.dart <-- package:opentelemetry/src/sdk/trace/propagation/w3c_trace_context_propagator.dart#L11 @@
-  class W3CTraceContextPropagator implements TextMapPropagator<dynamic>
// Removing a class is a major change.

-     void inject(Context context, dynamic carrier, TextMapSetter<dynamic> setter)
//    Removing a method is a major change.

-     W3CTraceContextPropagator W3CTraceContextPropagator()
//    Removing a constructor is a major change.

-     static RegExp get traceParentHeaderRegEx
//    Removing a field is a major change.

-     Context extract(Context context, dynamic carrier, TextMapGetter<dynamic> getter)
//    Removing a method is a major change.

      From dart:core/object.dart
-     Type get runtimeType
//    Removing a field is a major change.
-     int get hashCode
//    Removing a field is a major change.
-     bool ==(Object other)
//    Removing a method is a major change.
-     dynamic noSuchMethod(Invocation invocation)
//    Removing a method is a major change.
-     String toString()
//    Removing a method is a major change.

```

```diff
@@ opentelemetry/sdk.dart <-- package:opentelemetry/src/sdk/trace/span_context.dart#L12 @@
-  class SpanContext implements SpanContext
// Removing a class is a major change.

-     bool get isRemote
//    Removing a field is a major change.

-     SpanContext SpanContext.remote(TraceId _traceId, SpanId _spanId, int _traceFlags, TraceState _traceState)
//    Removing a constructor is a major change.

-     int get traceFlags
//    Removing a field is a major change.

-     SpanContext SpanContext(TraceId _traceId, SpanId _spanId, int _traceFlags, TraceState _traceState)
//    Removing a constructor is a major change.

-     SpanId get spanId
//    Removing a field is a major change.

-     TraceId get traceId
//    Removing a field is a major change.

-     bool get isValid
//    Removing a field is a major change.

-     SpanContext SpanContext.invalid()
//    Removing a constructor is a major change.

-     TraceState get traceState
//    Removing a field is a major change.

      From dart:core/object.dart
-     bool ==(Object other)
//    Removing a method is a major change.
-     int get hashCode
//    Removing a field is a major change.
-     String toString()
//    Removing a method is a major change.
-     dynamic noSuchMethod(Invocation invocation)
//    Removing a method is a major change.
-     Type get runtimeType
//    Removing a field is a major change.

```

Showing results for 9715c06274bf8331eab7742e323b769da933bc75
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.2
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/opentelemetry-dart/9715c06274bf8331eab7742e323b769da933bc75).

File: test/unit/api/span_context_test.dart [hunk 1]
```diff
@@ -6,6 +6,21 @@ import 'package:opentelemetry/api.dart' as api;
 import 'package:test/test.dart';
 
 void main() {
+  test('spanContext getters', () {
```
Comment (michaelyeager-wf) [inline test/unit/api/span_context_test.dart:9]: This test was present for the removed version of SpanContext but not for the API version.  I've moved it here for consistency.

**Comment by michaelyeager-wf** [comment]
We've decided to include this in a future version with other breaking changes.

**Pull request closed without merging**